### ISSUE TITLE
[v4] Text is cropped for buttons inside bottomsheet on IOS only

### ISSUE THREAD

**Issue by kav** [gorhom/react-native-scroll-bottom-sheet#1867]
# Bug
Buttons appearing inside a bottom sheet (on IOS) get cropped off. This behavior is random and does not occur every time, the best way to reproduce is to constantly refresh the app. This happens in the simulator and on my ios device.
 
<img alt="Screen Shot 2022-12-07 at 9 08 23 AM" width="324" src="https://user-images.githubusercontent.com/15200569/206202498-ef77ebb0-bf18-458a-a9df-f79a692c4bc9.png">

Another example:

<img alt="Screen Shot 2022-12-07 at 9 20 26 AM" width="286" src="https://user-images.githubusercontent.com/15200569/206203365-373eb840-ea40-45b0-9f7a-1edea9909768.png">

## Environment info
Library	Version
@gorhom/bottom-sheet	4.6.0
react-native	0.74.2
react-native-reanimated	3.10.1
react-native-gesture-handler	2.16.1
## Steps To Reproduce
1. Create a simple bottomsheet from https://gorhom.github.io/react-native-bottom-sheet/usage
2. Add a button
3. Add the following style to the content container:

```
  contentContainer: {
    flex: 1,
    alignItems: 'center',
  }
```

4. Continue refreshing the app until the text gets cropped off. E.g.

<img alt="Screen Shot 2022-12-07 at 9 18 07 AM" width="245" src="https://user-images.githubusercontent.com/15200569/206202742-cd276b53-7e3e-43e7-b785-aa692c88bb39.png">

Expected behavior:

Text does not get cut off:

<img alt="Screen Shot 2022-12-07 at 9 21 38 AM" width="299" src="https://user-images.githubusercontent.com/15200569/206203582-bfd5b172-59ad-4081-8a39-371800bf34c0.png">

## Reproducible sample code
1. Open https://snack.expo.dev/@jeannei91/bottom-sheet-cropped-button
2. Switch to the ios view
3. Reload the app until the button is cropped

Note this was also reported in 2021 ( https://github.com/gorhom/react-native-bottom-sheet/issues/645 ) but the issue was closed
And again in 2022 https://github.com/gorhom/react-native-bottom-sheet/issues/1218 but again the issue was closed

**Comment by kav** [comment]
Still a problem.

**Comment by ArseniiLyzenko** [comment]
+1 victim of this bug

**Comment by Darnell-Chen** [comment]
Try wrapping everything inside BottomSheet with BottomSheetView (as shown in docs) or ScrollView;

```
import React, { useCallback, useMemo, useRef } from 'react';
import { View, Text, StyleSheet, Button, TextInput } from 'react-native';
import BottomSheet, { BottomSheetView } from '@gorhom/bottom-sheet';
import { ScrollView } from 'react-native-gesture-handler';

const ManualForm = () => {
  // ref
  const bottomSheetRef = useRef<BottomSheet>(null);

  // callbacks
  const handleSheetChanges = useCallback((index: number) => {
    console.log('handleSheetChanges', index);
  }, []);

  // renders
  return (
    <View style={styles.container}>
      <BottomSheet
        ref={bottomSheetRef}
        onChange={handleSheetChanges}
        snapPoints={['5%', '25%', '50%', '75%', '90%']}
      >
        <BottomSheetView style={styles.contentContainer}>
        <Button title="Button"/>
        </BottomSheetView>
        <Button title="Cropped Button"/>
      </BottomSheet>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 24,
    backgroundColor: 'grey',
  },
  contentContainer: {
    flex: 1,
    alignItems: 'center',
  },
  textInput: {
    borderColor: 'lightgray',
    borderWidth: 2,
    padding: 5,
    fontSize: 20
  }
});


export default ManualForm;
```

## Result
![IMG_1862](https://github.com/user-attachments/assets/335b2e15-51ec-4693-9f87-4928fca3dbed)


In the example above, the Button that's outside of BottomSheetView will be cut off.

**Comment by Darnell-Chen** [comment]
From my tests, it only appears to be the case in some scenarios. My solution was to just use another view instead of <ButtonSheetView> since it seems to be a problem within that specific View component.

```
import React, { useCallback, useMemo, useRef } from 'react';
import { View, Text, StyleSheet, Button, TextInput } from 'react-native';
import BottomSheet, { BottomSheetView } from '@gorhom/bottom-sheet';
import { ScrollView } from 'react-native-gesture-handler';

const ManualForm = () => {
  // ref
  const bottomSheetRef = useRef<BottomSheet>(null);

  // callbacks
  const handleSheetChanges = useCallback((index: number) => {
    console.log('handleSheetChanges', index);
  }, []);

  return (
    <View style={styles.container}>
      <BottomSheet
        ref={bottomSheetRef}
        onChange={handleSheetChanges}
        snapPoints={['5%', '25%', '50%', '75%', '90%']}
      >
        <BottomSheetView style={styles.contentContainer}>
          <Button title="a Button"/>
        </BottomSheetView>

        <View>
          <Button title="In Regular View"/>
        </View>
        <ScrollView>
        <Button title="In ScrollView"/>
        </ScrollView>
      </BottomSheet>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 24,
    backgroundColor: 'grey',
  },
  contentContainer: {
    flex: 1,
    alignItems: 'center',
  },
  textInput: {
    borderColor: 'lightgray',
    borderWidth: 2,
    padding: 5,
    fontSize: 20
  }
});

export default ManualForm;
```

<img src="https://github.com/user-attachments/assets/3aef43be-be16-4e00-9a24-ff940d249e1e" width="300" height="500" />

In the example above, neither ScrollView nor View exhibit the same sort of issue. It might also be possible to not wrap the items <BottomSheet> with a View component.

*** This was tested on Expo Go

**Comment by Episodex** [comment]
I have the same problem still on 4.6.4. Changing `BottomSheetView` to `View` doesn't seem to do anything in my case. It's only on iOS and sometimes it crops just one button, sometimes more. It's kind of a show stopper unfortunately.

Edit: I noticed that v4 is not maintained anymore, so I installed 5.0.0-alpha.11 instead. The problem persisted, but when I changed `enableDynamicSizing` to true (and added back `BottomSheetView`) it now works perfectly.
